model_analyzer/config/generate/optuna_run_config_generator.py

```diff
@@ -0,0 +1,302 @@
+    ) -> ModelRunConfig:
+        default_model_config_variant = (
+            BaseModelConfigGenerator.make_model_config_variant(
+                param_combo={},
+                model=model,
+                model_variant_name_manager=self._model_variant_name_manager,
+                c_api_mode=self._c_api_mode,
+            )
+        )
+
+        default_perf_analyzer_config = self._create_default_perf_analyzer_config(
+            model, default_model_config_variant.model_config
+        )
+
+        default_model_run_config = ModelRunConfig(
+            model.model_name(),
+            default_model_config_variant,
+            default_perf_analyzer_config,
+        )
+
+        return default_model_run_config
+
+    def _create_default_perf_analyzer_config(
+        self, model: ModelProfileSpec, model_config: ModelConfig
+    ) -> PerfAnalyzerConfig:
+        default_perf_analyzer_config = PerfAnalyzerConfig()
+        default_perf_analyzer_config.update_config_from_profile_config(
+            model_config.get_field("name"), self._config
+        )
+
+        default_concurrency = self._calculate_default_concurrency(model_config)
+
+        perf_config_params = {
+            "batch-size": DEFAULT_BATCH_SIZES,
+            "concurrency-range": default_concurrency,
+        }
+        default_perf_analyzer_config.update_config(perf_config_params)
+
+        default_perf_analyzer_config.update_config(model.perf_analyzer_flags())
+
+        return default_perf_analyzer_config
+
+    def _calculate_default_concurrency(self, model_config: ModelConfig) -> int:
+        default_max_batch_size = model_config.max_batch_size()
+        default_instance_count = model_config.instance_group_count(
+            system_gpu_count=len(self._gpus)
+        )
+        default_concurrency = 2 * default_max_batch_size * default_instance_count
+
+        return default_concurrency
+
+    def _create_model_run_config(
+        self,
+        model: ModelProfileSpec,
+        model_config_variant: ModelConfigVariant,
+    ) -> ModelRunConfig:
+        perf_analyzer_config = self._create_perf_analyzer_config(
+            model.model_name(), model, self._concurrency
+        )
+        model_run_config = ModelRunConfig(
+            model.model_name(), model_config_variant, perf_analyzer_config
+        )
+
+        return model_run_config
+
+    def _create_perf_analyzer_config(
+        self,
+        model_name: str,
+        model: ModelProfileSpec,
+        concurrency: int,
+    ) -> PerfAnalyzerConfig:
+        perf_analyzer_config = PerfAnalyzerConfig()
+
+        perf_analyzer_config.update_config_from_profile_config(model_name, self._config)
+
+        perf_config_params = {"batch-size": 1, "concurrency-range": concurrency}
+        perf_analyzer_config.update_config(perf_config_params)
+
+        perf_analyzer_config.update_config(model.perf_analyzer_flags())
+        return perf_analyzer_config
+
+    def _print_debug_logs(
```

Comment:
Any reason to create a function that isn't implemented or called? You can create it if/when it is time to implement.

Reply:
If this code was going into main then this would be bad form. This is just a placeholder so I can keep the TODO in the right location (in case someone else picks up this work).

Reply:
is the TODO going to be resolved before this is merged into main?

Reply:
Yes, this will be done before it goes into main